pkg/runtime/types.go

```diff
@@ -120,26 +125,259 @@ type Unknown struct {
 // TODO: Make this object have easy access to field based accessors and settors for
 // metadata and field mutatation.
 type Unstructured struct {
-	TypeMeta `json:",inline"`
-
-	// Name is populated from metadata (if present) upon deserialization
-	Name string
-
 	// Object is a JSON compatible map with string, float, int, []interface{}, or map[string]interface{}
 	// children.
 	Object map[string]interface{}
 }
 
+func getNestedField(obj map[string]interface{}, fields ...string) interface{} {
```

Comment:
This is more of a follow up, but I'm pretty sure we have a library somewhere in our godeps that does this.  Just not coming to mind off the top of my head.  No change necessary here, but if we extend this we should switch.